vfs/cache.go

```diff
@@ -270,12 +270,63 @@ func (c *cache) exists(name string) bool {
 	if err != nil {
 		return false
 	}
-	if fi.IsDir() {
+	// checks for non-regular files (e.g. directories, symlinks, devices, etc.)
+	if !fi.Mode().IsRegular() {
 		return false
 	}
 	return true
 }
 
+// renames the file in cache
+func (c *cache) rename(name string, newName string) bool {
```

Comment:
It would be more go style to return an annotated error, than a boolean here I think...